### ISSUE TITLE
[3.3.5] worldserver.exe crash

### ISSUE THREAD

**Issue by ChiefMcClane** [TrinityCore/TrinityCore#24873]
**Description:**

I have pulled the latest commit from the 3.3.5 branch. The authserver and worldserver were compiled cleanly and without errors. But when I try to start the worldserver it crashes after this:

...
VMap data directory is: ./vmaps
Loading Trinity strings...
>> Loaded 967 trinity strings in 11 ms
Initialize data stores...

Crashlog see following link. 
https://pastebin.com/a5XGkAG6

**Steps to reproduce the problem:**

1. Compile Server on Debug
2. Extract and copy vmaps, mmaps, maps and dbc
3. Manage Database
4. Run auth and worldserver

**Branch(es):**

3.3.5

**Operating system:** 

Windows 10

**TC rev. hash/commit:**

dc41aae0427faecef57b831f3a2ed735a9369dd7

**Comment by jackpoz** [comment]
where did you get the 3.3.5 client used to extract dbcs ? some private server link or ?

**Comment by shamage** [comment]
compilation in Visual Studio went smoothly? 

 I've just done the set : 

TrinityCore rev. dc41aae0427f 2020-06-22 15:02:50 +0200 (3.3.5 branch) (Win64, RelWithDebInfo, Static)
Using SSL version: OpenSSL 1.1.1g  21 Apr 2020 (library: OpenSSL 1.1.1g  21 Apr 2020)
Using Boost version: 1.72.0
Using MySQL version: 50730
Using CMake version: 3.17.3
Compiled on: Windows 10.0.18363
Automatic database updates are enabled for the following databases: Auth, Characters, World
Worldserver listening connections on port 8085
Realmlist (Realm Id: 1) configured in port 8085
VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
MMAPs status: Disabled
maps directory located in ./maps. Total size: 251987687 bytes
vmaps directory located in ./vmaps. Total size: 658130721 bytes
Using enUS DBC Locale as default. All available DBC locales: enUS
Using World DB: TDB 335.20061

everything works without any problems  

@ChiefMcClane : try to see if you made a mistake somewhere

**Comment by ChiefMcClane** [comment]
> where did you get the 3.3.5 client used to extract dbcs ? some private server link or ?

@jackpoz 

I can't remember where I got the client. It's either an old version of Firestorm or it's been downloaded from the Internet somewhere.



> compilation in Visual Studio went smoothly?
> 
> I've just done the set :
> 
> TrinityCore rev. [dc41aae](https://github.com/TrinityCore/TrinityCore/commit/dc41aae0427faecef57b831f3a2ed735a9369dd7) 2020-06-22 15:02:50 +0200 (3.3.5 branch) (Win64, RelWithDebInfo, Static)
> Using SSL version: OpenSSL 1.1.1g 21 Apr 2020 (library: OpenSSL 1.1.1g 21 Apr 2020)
> Using Boost version: 1.72.0
> Using MySQL version: 50730
> Using CMake version: 3.17.3
> Compiled on: Windows 10.0.18363
> Automatic database updates are enabled for the following databases: Auth, Characters, World
> Worldserver listening connections on port 8085
> Realmlist (Realm Id: 1) configured in port 8085
> VMAPs status: Enabled. LineOfSight: 1, getHeight: 1, indoorCheck: 1
> MMAPs status: Disabled
> maps directory located in ./maps. Total size: 251987687 bytes
> vmaps directory located in ./vmaps. Total size: 658130721 bytes
> Using enUS DBC Locale as default. All available DBC locales: enUS
> Using World DB: TDB 335.20061
> 
> everything works without any problems
> 
> @ChiefMcClane : try to see if you made a mistake somewhere

@shamage 

Yes I meant with Compiling Visual Studio. It all went well. 

The points you made, I have them on me, too. One caution is that I'm using a slightly older version of OpenSSL. Also I didn't enable the autoupdate update function, because I rebuilt the server. I even have exactly the same size of the folders

**Comment by Ovahlord** [comment]
Get a clean client. What you have there looks alot like either a corruped or a modified client with custom titles which the core does not really like having

**Comment by ChiefMcClane** [comment]
@Aokromes why closed? That is Not helpful!

**Comment by Jinnaix** [comment]
Because its pretty sure related to a corrupted/modified client and therefore nothing where Trinitycore Devs can help with.

**Comment by ChiefMcClane** [comment]
@Jinnaix  @Ovahlord 

Okay. I see that. Then why don't you write about how to get a clean client that works for Trinity? You didn't get that from Santa Claus either, did you?

**Comment by Rothend** [comment]
Any unmodified 3.3.5a client works. It's not TC's job to provide it, there's plenty around offered by blizzlike servers.

**Comment by Jinnaix** [comment]
You dont need to be cheeky.
Trinitycore provides a server framework, not a client. Sharing Blizzard`s copyright protected client isnt a thing we do. Google is your friend.

**Comment by illfated** [comment]
@ChiefMcClane : From my experience during the last 5-6 years, using backup copies of my original 2010 3.3.5a (12340) game client provided the most stability and as few client issues as possible for me when testing various 3.3.5 server sources.

Also from my experience, when I sometimes tested a couple of different downloads from torrent or Pirate Bay, it was very much up to me only to decipher if the uploaded content was safe enough to use with most server sources. I still remember seeing something similar to your crash log if the game client was from a dubious source, knowing that the DBC files and maps need to be extracted from the game client. Unless the game client file content was correct, I did have similar issues to your crash log, because the server did not want to load corrupt or incomplete DBC / map contents.

TL;DR: in my opinion, it is an individual, personal responsibility to have a clean copy of a 3.3.5a game client to make any personal open-source server work well with the game client, because of the shared file content between server and game client.
